nomad/job_endpoint_hook_implicit_identities_test.go

```diff
@@ -248,27 +296,248 @@ func Test_jobImplicitIdentitiesHook_Mutate_consul_service(t *testing.T) {
 			},
 			expectedOutputJob: &structs.Job{
 				TaskGroups: []*structs.TaskGroup{{
-					Name: "group",
 					Constraints: []*structs.Constraint{
-						implicitIdentityClientVersionConstraint()},
+						implicitIdentityClientVersionConstraint(),
+					},
 					Tasks: []*structs.Task{{
-						Name:      "web-task",
-						Templates: []*structs.Template{{}},
-						Identities: []*structs.WorkloadIdentity{{
-							Name:     "consul_default",
+						Name:   "web-task",
+						Consul: &structs.Consul{},
+						Identities: []*structs.WorkloadIdentity{
+							{
+								Name:     "consul_default",
+								Audience: []string{"consul.io"},
+							},
+						},
+					}},
+				}},
+			},
+		},
+		{
+			name: "consul block in task group without identity block",
+			inputJob: &structs.Job{
+				TaskGroups: []*structs.TaskGroup{{
+					Consul: &structs.Consul{},
+					Tasks: []*structs.Task{{
+						Name: "web-task",
+					}},
+				}},
+			},
+			inputConfig: &Config{
+				ConsulConfigs: map[string]*config.ConsulConfig{
+					structs.ConsulDefaultCluster: {
+						TaskIdentity: &config.WorkloadIdentityConfig{
 							Audience: []string{"consul.io"},
-						}},
+						},
+					},
+				},
+			},
+			expectedOutputJob: &structs.Job{
+				TaskGroups: []*structs.TaskGroup{{
+					Consul: &structs.Consul{},
+					Constraints: []*structs.Constraint{
+						implicitIdentityClientVersionConstraint(),
+					},
+					Tasks: []*structs.Task{{
+						Name: "web-task",
+						Identities: []*structs.WorkloadIdentity{
+							{
+								Name:     "consul_default",
+								Audience: []string{"consul.io"},
+							},
+						},
 					}},
 				}},
 			},
 		},
 		{
-			name: "no mutation for templates when no task identity is configured",
+			name: "consul block in task and task consul identity block",
 			inputJob: &structs.Job{
 				TaskGroups: []*structs.TaskGroup{{
 					Tasks: []*structs.Task{{
-						Name:      "web-task",
-						Templates: []*structs.Template{{}},
+						Name:   "web-task",
+						Consul: &structs.Consul{},
+						Identities: []*structs.WorkloadIdentity{
+							{
+								Name:     "consul_default",
+								Audience: []string{"consul.io"},
+							},
+						},
+					}},
```

Comment:
Maybe the test input from jobs should have a different audience so we can ensure that we're using the identity specified by the job author rather than the implicit one when available?

Reply:
Great shout! This uncovered a bug in the handling of task vs. task group priorities. The update to the test and function is commit: https://github.com/hashicorp/nomad/pull/25298/commits/8695d3706833a604ba471ce23bb300147b4a7ddd